bentoml/_internal/runner/runner.py

```diff
@@ -183,238 +140,179 @@ def get_runnable_methods(cls) -> t.List[RunnableMethod]:
+    async def async_run(self, *args: t.Any, **kwargs: t.Any) -> t.Any:
+        return await self.runner.async_run_method(self.method_name, *args, **kwargs)
 
 
-class RunnerResourceConfig:
-    ...
+@attr.define(frozen=True)
+class Resource:
+    cpu: int = attr.field()
+    nvidia_gpu: int = attr.field()
+    custom_resources: t.Dict[str, t.Union[float, int]] = attr.field(factory=dict)
 
 
-class Runner:
-    """
-    TODO: add docstring
-    """
-
-    def __init__(
-        self,
-        runnable_class,
-        init_params,
-        name,
-        strategy,
-        models,
-        cpu,
-        nvidia_gpu,
-        custom_resources,
-        max_batch_size,
-        max_latency_ms,
-        runnable_method_configs: t.Dict[str, t.Dict[str, str | int]],
-    ) -> None:
-        self._runnable_class = runnable_class
-        self._runnable_init_params = init_params
-        self._name = name
-        self._strategy = strategy
-        self._model = models
-        self._resource_config = RunnerResourceConfig(cpu, nvidia_gpu, custom_resources)
-
-        runnable_method_configs = runnable_method_configs or {}
-        for runnable_method in self.runnable_class.get_runnable_methods():
-            method_config = runnable_method_configs.get(runnable_method.name, {})
-            setattr(
-                self,
-                runnable_method.name,
-                RunnerMethod(
-                    self,
-                    runnable_method,
-                    max_batch_size=method_config.get("max_batch_size", max_batch_size),
-                    max_latency_ms=method_config.get("max_batch_size", max_latency_ms),
-                ),
-            )
+@attr.define(frozen=True)
+class RunnerMethodConfig:
+    max_batch_size: int = attr.field(default=1000)
+    max_latency_ms: int = attr.field(default=10000)
 
-        self._runner_app_client: "RunnerClient" = None
-        self._runnable: Runnable = None
 
-    @property
-    def name(self):
-        return self._name
+@attr.define
+class RunnerHandle:
+    _runnable: Runnable | None = attr.field(init=False, default=None)
+    _runner_client: RemoteRunnerClient | None = attr.field(init=False, default=None)
 
-    @property
-    def strategy(self):
-        return self._strategy
+    def init_local(self, runner: Runner):
+        logger.warning("for debugging and testing only")  # if not called from RunnerApp
+        if self._runner_client:
+            raise BentoMLException("TODO: ..")
+        if self._runnable:
+            logger.warning("re creating runnable")
 
-    @property
-    def models(self):
-        return self._models
+        self._runnable = runner.runnable_class()
 
-    @property
-    def resource_config(self):
-        # look up configuration
-        # else return self._resource_config
-        ...
+    def destroy_local(self):
+        if not self._runnable:
+            logger.warning("local runnable not found")
+        else:
+            del self._runnable
```

Comment:
Is `destroy_local()` necessary? Do we expect the `RunnerHandle` class to be re-init? Same for `destroy_client()` below.

Reply:
it will be used in the RunnerApp only, not meant to be invoke by users